eth/tracers/tracers.go

```diff
@@ -18,14 +18,53 @@
 package tracers
 
 import (
+	"encoding/json"
 	"strings"
 	"unicode"
 
+	"github.com/ethereum/go-ethereum/core/vm"
 	"github.com/ethereum/go-ethereum/eth/tracers/internal/tracers"
 )
 
-// all contains all the built in JavaScript tracers by name.
-var all = make(map[string]string)
+// Tracer interface extends vm.EVMLogger and additionally
+// allows collecting the tracing result.
+type Tracer interface {
+	vm.EVMLogger
+	GetResult() (json.RawMessage, error)
+	// Stop terminates execution of the tracer at the first opportune moment.
+	Stop(err error)
+}
+
+var (
+	nativeTracers map[string]func() Tracer = make(map[string]func() Tracer)
+	jsTracers                              = make(map[string]string)
+)
+
+// RegisterNativeTracer makes native tracers which adhere
+// to the `EVMLogger` interface available to the rest of the codebase.
+// It is typically invoked in the `init()` function, e.g. see the `native/call.go`.
+func RegisterNativeTracer(name string, ctor func() Tracer) {
+	nativeTracers[name] = ctor
+}
+
+// New returns a new instance of a tracer,
+// 1. If 'code' is the name of a registered native tracer, then that tracer
+//   is instantiated and returned
+// 2. If 'code' is the name of a registered js-tracer, then that tracer is
+//   instantiated and returned
+// 3. Otherwise, the code is interpreted as the js code of a js-tracer, and
+//   is evaluated and returned.
+func New(code string, ctx *Context) (Tracer, error) {
+	// Resolve native tracer
+	if fn, ok := nativeTracers[code]; ok {
+		return fn(), nil
+	}
+	// Resolve js-tracers by name and assemble the tracer object
+	if tracer, ok := jsTracers[code]; ok {
+		code = tracer
+	}
+	return newJsTracer(code, ctx)
```

Comment:
Anotherb way to do this.. would be to have a list of frameworks. 
Both js and native does
```
RegisterTracingFramework( lookupTracer)
```
And here, we do basically
```
for _ , framework := range frameworks{
   if tracer, ok := framework.loadTracer(code, ctx); ok{
     return tracer, nil
  }
  return nil, errors.New("no tracer found")
}
```
Then tracers would be totally unaware of the js subpackage, which under the hood could implement wildcard matching. 
The drawback would be that the order in which they register would matter: we need the native one to register first, since otherwise `nativeCallTracer` will be picked up by the wildcard matcher. 
If we don't like that, we could do `RegisterTracingFramework( lookupFn func ... , wildcard bool)`, and make sure to place a wildcard-matcher last. 
Doing this would also make it trivial to have two js-engines side by side, and if we just don't include the reference from geth/main, then it would be omitted from the binary.